src/discussions/data/thunks.js

```diff
@@ -33,3 +36,29 @@ export function fetchCourseConfig(courseId) {
     }
   };
 }
+
+/**
+ * Fetches user threads for the course
+ * @param {string} courseId The course ID for the course to fetch threads for.
+ * @param {string} courseId The author username to fetch threads for.
+ * @returns {(function(*): Promise<void>)|*}
+ */
+export function fetchUserThreads(courseId, author) {
+  return async (dispatch) => {
+    try {
+      const options = {
+        author,
+      };
+      dispatch(fetchConfigRequest());
+      const data = await getThreads(courseId, options);
+      dispatch(fetchConfigSuccess({ userHasThreads: data.results.length > 0 }));
+    } catch (error) {
+      if (getHttpErrorStatus(error) === 403) {
+        dispatch(fetchConfigDenied());
+      } else {
+        dispatch(fetchConfigFailed());
+      }
+      logError(error);
+    }
+  };
+}
```

Comment:
There is no need for this, you can just use `fetcThreads(courseId, {author:username})`. There is no need to discard the results after that, if you use the existing method, then it will pre-load the data.

Reply:
@xitij2000 I tried that already, but after the redirection to that store get updated again with the params , I'll tried again making it working.

Reply:
@xitij2000 PR updated

Reply:
@xitij2000 looks like the tests coverage should be accepted now.
https://app.codecov.io/gh/openedx/frontend-app-discussions/compare/khchine5%2FBB-5598...master